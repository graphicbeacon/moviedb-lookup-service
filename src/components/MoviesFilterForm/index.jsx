import React, { useState } from 'react';

// Receives search term and returns it
const MoviesFilterForm = ({ onUpdate = () => {}, ...inputProps }) => {
  // Default state for form input
  const [search, setSearch] = useState('');
  const onSubmitHandler = e => {
    e.preventDefault();
    onUpdate(search);
  };

  return (
    <form
      data-testid="searchForm"
      onSubmit={onSubmitHandler}
      className="mt-5 mb-5"
    >
      <div className="input-group">
        <input
          type="search"
          data-testid="searchInput"
          className="form-control form-control-lg"
          value={search}
          onChange={e => setSearch(e.target.value)}
          {...inputProps}
        />
        <div className="input-group-append">
          <button
            data-testid="searchSubmit"
            className="btn btn-primary"
            type="submit"
          >
            Search
          </button>
        </div>
      </div>
    </form>
  );
};

export default MoviesFilterForm;
