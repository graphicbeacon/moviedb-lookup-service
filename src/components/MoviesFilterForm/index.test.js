import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, fireEvent, getByTestId } from 'react-testing-library';

import MoviesFilterForm from './index';

describe('<MoviesFilterForm />', () => {
  test('should match generated snapshot', () => {
    // Arrange, Act
    const tree = TestRenderer.create(<MoviesFilterForm />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  test('should trigger form update event', () => {
    // Arrange
    const onUpdateFn = jest.fn();
    const { container } = render(<MoviesFilterForm onUpdate={onUpdateFn} />);
    const form = getByTestId(container, 'searchForm');
    const button = getByTestId(container, 'searchSubmit');

    // Act, Assert
    fireEvent.submit(form);
    expect(onUpdateFn.mock.calls.length).toBe(1);

    // Act, Assert
    fireEvent.click(button);
    expect(onUpdateFn.mock.calls.length).toBe(2);
  });

  test('should update input with search term', () => {
    // Arrange
    const onSubmitFn = jest.fn();
    const { container } = render(<MoviesFilterForm onUpdate={onSubmitFn} />);
    const input = getByTestId(container, 'searchInput');

    // Act
    fireEvent.change(input, { target: { value: 'search term' } });

    // Assert
    expect(input.value).toBe('search term');
  });

  // TODO: Should prevent default submit event

  test('Should have empty method if submit handler not specified', () => {
    // Arrange, Act
    const tree = TestRenderer.create(<MoviesFilterForm />).toJSON();

    // Assert
    expect(tree.props.onSubmit).toBeDefined();
  });

  test('should have placeholder with instructions', () => {
    // Arrange, Act
    const { container } = render(
      <MoviesFilterForm placeholder="Enter search and submit" />
    );
    const input = getByTestId(container, 'searchInput');

    // Assert
    expect(input.placeholder).toEqual('Enter search and submit');
  });
});
