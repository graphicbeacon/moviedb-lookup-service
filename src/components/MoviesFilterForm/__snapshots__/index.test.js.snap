// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesFilterForm /> should match generated snapshot 1`] = `
<form
  className="mt-5 mb-5"
  data-testid="searchForm"
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <input
      className="form-control form-control-lg"
      data-testid="searchInput"
      onChange={[Function]}
      type="search"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-primary"
        data-testid="searchSubmit"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</form>
`;
